allen-cell-animated/timelapse-colorizer
# feature: Multithread texture packing

Problem
=======
*Estimated review size: small, 5-10 minutes*

Closes #422, use workers when packing textures. This skips a VERY minor slowdown on the main UI-handling thread when the texture data was being processed.

On average (n=18 for new vs. old loader) the exploratory datasets finish processing about 30-150ms faster, but there's a lot of noise in the time trials I collected based on network and computer performance. S3 may have also been caching and serving files faster because of the repeated requests for them.

![image](https://github.com/user-attachments/assets/e75759fb-929e-4e52-9ef9-5b404eac4ba7)

Solution
========
- Perform the texture packing step on the worker instead of the main thread.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-434/
2. Open any of the datasets and verify that the feature data loads and looks correct.
